src/esp/metadata/managers/SensorAttributesManager.cpp

```diff
@@ -0,0 +1,259 @@
+          AbstractAttributesManager("Sensor Attributes", "sensor_config.json") {
+  // create constructor maps for various types of sensors mapping to their
+  // attributes
+  sensorTypeConstructorMap_["AudioSensorAttributes"] =
+      &SensorAttributesManager::createSensorAttributes<
+          attributes::AudioSensorAttributes>;
+  sensorTypeConstructorMap_["CameraSensorAttributes"] =
+      &SensorAttributesManager::createSensorAttributes<
+          attributes::CameraSensorAttributes>;
+  sensorTypeConstructorMap_["CustomSensorAttributes"] =
+      &SensorAttributesManager::createSensorAttributes<
+          attributes::CustomSensorAttributes>;
+  sensorTypeConstructorMap_["EquirectangularSensorAttributes"] =
+      &SensorAttributesManager::createSensorAttributes<
+          attributes::EquirectangularSensorAttributes>;
+  sensorTypeConstructorMap_["FisheyeSensorAttributes"] =
+      &SensorAttributesManager::createSensorAttributes<
+          attributes::FisheyeSensorAttributes>;
+
+  // create copy constructor map entries for each type of Sensor Attributes
+  this->copyConstructorMap_["AudioSensorAttributes"] =
+      &SensorAttributesManager::createObjCopyCtorMapEntry<
+          attributes::AudioSensorAttributes>;
+  this->copyConstructorMap_["CameraSensorAttributes"] =
+      &SensorAttributesManager::createObjCopyCtorMapEntry<
+          attributes::CameraSensorAttributes>;
+  this->copyConstructorMap_["CustomSensorAttributes"] =
+      &SensorAttributesManager::createObjCopyCtorMapEntry<
+          attributes::CustomSensorAttributes>;
+  this->copyConstructorMap_["EquirectangularSensorAttributes"] =
+      &SensorAttributesManager::createObjCopyCtorMapEntry<
+          attributes::EquirectangularSensorAttributes>;
+  this->copyConstructorMap_["FisheyeSensorAttributes"] =
+      &SensorAttributesManager::createObjCopyCtorMapEntry<
+          attributes::FisheyeSensorAttributes>;
+
+}  // SensorAttributesManager ctor
+
+AbstractSensorAttributes::ptr
+SensorAttributesManager::createAttributesFromSensorSpecInternal(
+    const sensor::SensorSpec::ptr& sensorSpec,
+    bool registerTemplate) {
+  // Get attributes class name from sensorSpec
+  const std::string& sensorAttrClassName =
+      SensorAttributesManager::SenssorAttrsTypeNamesMap.at(
+          sensorSpec->sensorSubType);
+  CORRADE_ASSERT(!sensorAttrClassName.empty(),
+                 "Unknown SensorAttributes class for SensorSubType enum value :"
+                     << static_cast<int32_t>(sensorSpec->sensorSubType),
+                 nullptr);
+  // Build and initialize appropriate class of Attributes based on class name
+  auto sensorAttributes =
+      this->initNewObjectInternal(sensorAttrClassName, false);
+  // Populate attributes from sensorSpec
+  sensorAttributes->populateWithSensorSpec(sensorSpec);
+  // TODO : Rename attributes to appropriate name and register
+  // Build name using more than just spec uuid?
+  std::string newAttrHandle = sensorSpec->uuid;
```

Comment:
Probably need some de-duplication system for unique naming. Spec will go away, so this doesn't need to be tied to that field. Naming could be a part of registering if it doesn't come from a filename for example.

Reply:
Yeah, postCreateRegister below does this (makes sure the name is unique - uuid becomes uuid_XXXX where XXXX is a unique number.